src/main/java/org/spongepowered/api/text/translation/dictionary/TranslationDictionary.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.translation.dictionary;
+
+import org.spongepowered.api.text.translation.locale.Locales;
+
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * Represents a TranslationDictionary for a particular subject.
+ *
+ * <p>Dictionaries take a given string key and return a translated result.</p>
+ */
+public interface TranslationDictionary {
+
+    /**
+     * Gets the entry for the specified key in this dictionary
+     * for the {@link #getDefaultLocale() default locale}.
+     *
+     * @param key The key whose associated value is to be returned
+     * @return The string value for {@code key}, if present,
+     *     {@link Optional#empty()} otherwise
+     */
+    default Optional<String> get(String key) {
+        return this.get(key, this.getDefaultLocale());
+    }
+
+    /**
+     * Gets the entry for the specified key for the specified
+     * {@link Locale}.
+     *
+     * @param key The key whose associated value is to be returned
+     * @param locale The locale under which the value should be
+     *     obtained in
+     * @return The string value for {@code key}, if present,
+     *     {@link Optional#empty()} otherwise
+     */
+    Optional<String> get(String key, Locale locale);
+
+    /**
+     * Gets the default {@link Locale} to be used with this dictionary.
+     *
+     * @return The default locale
+     */
+    default Locale getDefaultLocale() {
+        return Locales.DEFAULT;
```

Comment:
In my opinion you should be able to override this per dictionary. Eg, if a plugin is Spanish, and the default language is Spanish, you'd want that to be the fallback as it'd be the most complete and accurate. However I'm assuming DEFAULT is system default.  What if the default is Chinese and the plugin doesn't provide a Chinese translation?